e2e/js_image/js_image_layer.bzl

```diff
@@ -0,0 +1,239 @@
+
+    manifest = default.files_to_run.runfiles_manifest
+    runfiles_dir = "/".join([ctx.attr.root, manifest.short_path.replace(manifest.basename, "")[:-1]])
+
+    files = depset(transitive = [default.files, default.default_runfiles.files])
+
+    for file in files.to_list():
+        destination = _runfile_path(ctx, file, runfiles_dir)
+        if _should_include(destination, ctx.attr.include, ctx.attr.exclude):
+            file_map[destination] = file
+
+    # executable and launcher should not go into runfiles directory so we add it to files here
+    files = depset([executable, launcher], transitive = [files])
+
+    symlinks = []
+
+    # NOTE: symlinks is different than root_symlinks. See: https://bazel.build/rules/rules#runfiles_symlinks for distinction between
+    # root_symlinks and symlinks and why they have to be handled differently.
+    for symlink in default.data_runfiles.symlinks.to_list():
+        destination = "/".join([runfiles_dir, ctx.workspace_name, symlink.path])
+        if not _should_include(destination, ctx.attr.include, ctx.attr.exclude):
+            continue
+        if hasattr(file_map, destination):
+            file_map.pop(destination)
+        info = PackageSymlinkInfo(
+            target = "/%s" % _runfile_path(ctx, symlink.target_file, runfiles_dir),
+            destination = destination,
+            attributes = {"mode": "0777"},
+        )
+        symlinks.append([info, symlink.target_file.owner])
+
+    for symlink in default.data_runfiles.root_symlinks.to_list():
+        destination = "/".join([runfiles_dir, symlink.path])
+        if not _should_include(destination, ctx.attr.include, ctx.attr.exclude):
+            continue
+        if hasattr(file_map, destination):
+            file_map.pop(destination)
+        info = PackageSymlinkInfo(
+            target = "/%s" % _runfile_path(ctx, symlink.target_file, runfiles_dir),
+            destination = destination,
+            attributes = {"mode": "0777"},
+        )
+        symlinks.append([info, symlink.target_file.owner])
+
+    return [
+        PackageFilegroupInfo(
+            pkg_dirs = [],
+            pkg_files = [
+                [PackageFilesInfo(
+                    dest_src_map = file_map,
+                    attributes = {},
+                ), ctx.label],
+            ],
+            pkg_symlinks = symlinks,
+        ),
+        DefaultInfo(files = files),
+    ]
+
+runfiles = rule(
+    implementation = _runfiles_impl,
+    attrs = {
+        "binary": attr.label(mandatory = True),
+        "root": attr.string(),
+        "include": attr.string(),
+        "exclude": attr.string(),
+    },
+)
+
+BUILD_TAR = """
+import os
+import sys
+import json
+import subprocess
+from rules_python.python.runfiles import runfiles
+
+r = runfiles.Create()
+build_tar_path = r.Rlocation("rules_pkg/pkg/private/tar/build_tar")
+
+manifest_path = next((argv.replace("--manifest=", "") for argv in sys.argv if argv.startswith("--manifest=")), None)
```

Comment:
Are the args always passed in a specific order? If so, the `argv[3]` or similar would suffice, removing the need to iterate through.

Reply:
it changes with attributes passed in.